engine/src/main/java/org/terasology/rendering/nui/layers/mainMenu/UniverseSetupScreen.java

```diff
@@ -140,30 +177,48 @@ public void set(String value) {
         );
 
         WorldSetupScreen worldSetupScreen = getManager().createScreen(WorldSetupScreen.ASSET_URI, WorldSetupScreen.class);
-        WidgetUtil.trySubscribe(this, "worldConfig", button ->
-                triggerForwardAnimation(worldSetupScreen)
-        );
+        WidgetUtil.trySubscribe(this, "worldConfig", button -> {
+            try {
+                if (!worlds.isEmpty() || !selectedWorld.isEmpty()) {
+                    worldSetupScreen.setWorld(context, findWorldByName());
+                    triggerForwardAnimation(worldSetupScreen);
+                } else {
+                    getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class).setMessage("Worlds List Empty!", "No world found to configure.");
+                }
+            } catch (UnresolvedWorldGeneratorException e) {
+                e.getMessage();
+            }
+        });
 
         WidgetUtil.trySubscribe(this, "addGenerator", button -> {
             addNewWorld(worldGenerator.getSelection());
-            List<String> worldOptions = Lists.newArrayList(worlds.keySet());
-            worldsDropdown.setOptions(worldOptions);
+            List<WorldSetupWrapper> worldOptions = worlds;
+            worldsDropdown.setOptions(worldNames());
             //triggerForwardAnimation(worldSetupScreen);
         });
 
         WorldPreGenerationScreen worldPreGenerationScreen = getManager().createScreen(WorldPreGenerationScreen.ASSET_URI, WorldPreGenerationScreen.class);
-        WidgetUtil.trySubscribe(this, "continue", button ->
-                triggerForwardAnimation(worldPreGenerationScreen)
-        );
+        WidgetUtil.trySubscribe(this, "continue", button -> {
+            if (!worlds.isEmpty()) {
+                try {
+                    worldPreGenerationScreen.setEnvironment(context);
+                    triggerForwardAnimation(worldPreGenerationScreen);
+                } catch (UnresolvedWorldGeneratorException e) {
+                    e.getMessage();
+                }
```

Comment:
What exactly does `e.getMessage` do?

(Alternatively, why are you retrieving the message and not actually printing it? If you're too damn lazy, just use e.printStackTrace(). Way shorter, and way more informative than a simple "NPE".)